Formula/envoy.rb

```diff
@@ -49,11 +35,13 @@ class Envoy < Formula
   # Starting with 1.21, envoy requires a full Xcode installation, not just
   # command-line tools. See envoyproxy/envoy#16482
   depends_on xcode: :build
-  depends_on macos: :catalina
+  # Build failure on macOS 10.15 due to error at source/extensions/filters/http/file_system_buffer/filter.cc:402:53:
+  # error: no viable constructor or deduction guide for deduction of template arguments of 'weak_ptr'.
+  depends_on macos: :big_sur
```

Comment:
We may want to backport https://github.com/envoyproxy/envoy/pull/22334 to avoid dropping Catalina support (though not too important as Ventura will be out soon so Homebrew's main support macOS versions will be for Big Sur and newer)

Reply:
Sure. cc. @cpakulski